src/sss_client/pam_sss.c

```diff
@@ -1207,6 +1209,9 @@ static int eval_response(pam_handle_t *pamh, size_t buflen, uint8_t *buf,
         p += len;
 
         --c;
+
+        free(env_item);
```

Comment:
Probably I miss something, but `man putenv`:
```
int putenv(char *string);

The string pointed to by string becomes part of the environment, so altering the string changes the environment.
```
so we shouldn't free it in case we return 'PAM_SUCCESS'?

The question what should we do in case we return 'PAM_BUF_ERR' is more tricky.